Dockerfile

```diff
@@ -1,27 +1,67 @@
-FROM google/cloud-sdk:alpine
+# ======================
+# Build target
+# ======================
+
+FROM alpine:3.10 AS build
 
 # Note: Latest version of helm may be found at:
 # https://github.com/kubernetes/helm/releases
 ENV HELM_VERSION="v2.14.1"
 
+RUN apk add --update \
+    bash \
+    gcc \
+    libffi-dev \
+    openssl-dev \
+    make \
+    musl-dev \
+    python3 \
+    python3-dev \
+    py3-pip
+
+# Install AWS cli
+RUN python3 -m venv /opt/aws \
+    && source /opt/aws/bin/activate \
+    && pip install awscli \
+    && deactivate
+
+# Install Azure cli
+RUN python3 -m venv /opt/az \
+    && source /opt/az/bin/activate \
+    && pip install azure-cli \
+    && deactivate
+
+# Install Helm
+RUN wget -q https://storage.googleapis.com/kubernetes-helm/helm-${HELM_VERSION}-linux-amd64.tar.gz -O - | tar -xzO linux-amd64/helm > /usr/local/bin/helm
+
+## ======================
+## Production target
+## ======================
+
+FROM google/cloud-sdk:alpine as prod
```

Comment:
Should we keep this as the production target? Maybe it's nicer if we make this in the build stage too?

Reply:
I tried but the Gcloud Python packages are split up into chunks so at this point it's pretty unclear to me how to get the right stuff installed.

Reply:
```
# Downloading gcloud package
RUN curl https://dl.google.com/dl/cloudsdk/release/google-cloud-sdk.tar.gz > /tmp/google-cloud-sdk.tar.gz

# Installing the package
RUN mkdir -p /usr/local/gcloud \
  && tar -C /usr/local/gcloud -xvf /tmp/google-cloud-sdk.tar.gz \
  && /usr/local/gcloud/google-cloud-sdk/install.sh

# Adding the package path to local
ENV PATH $PATH:/usr/local/gcloud/google-cloud-sdk/bin
```

```
RUN curl -sSL https://sdk.cloud.google.com | bash
```

```
FROM alpine:3.6

 RUN apk add --update \
 python \
 curl \
 which \
 bash

 RUN curl -sSL https://sdk.cloud.google.com | bash

 ENV PATH $PATH:/root/google-cloud-sdk/bin
```

Found some methods others are doing it